pss/outbox/outbox.go

```diff
@@ -51,20 +50,16 @@ type forwardFunction func(msg *message.Message) error
 // ErrOutboxFull is returned when a caller tries to enqueue a message and all slots are busy.
 var ErrOutboxFull = errors.New("outbox full")
 
-const defaultOutboxWorkers = 100
 const defaultMaxRetryTime = 10 * time.Minute
 
 // NewOutbox creates a new Outbox. Config must be provided. IF NumWorkers is not providers, default will be used.
 func NewOutbox(config *Config) *Outbox {
-	if config.NumWorkers == 0 {
-		config.NumWorkers = defaultOutboxWorkers
-	}
 	outbox := &Outbox{
 		forwardFunc:  config.Forward,
 		queue:        make([]*outboxMsg, config.NumberSlots),
 		slots:        make(chan int, config.NumberSlots),
 		process:      make(chan int),
-		numWorkers:   config.NumWorkers,
+		numWorkers:   config.NumberSlots,
```

Comment:
Not sure why we need this anymore, we could just use `len(queue)` - also the naming is confusing - is it slots or workers? I thought we are removing workers.

Reply:
Workers stay, what we remove is to have a huge queue bigger than workers, now we have one slot / one worker. I used config.NumberSlots for convenince, if not I need to first initialize the queue outside the struct initialization (anyway I will change that).